gcp/rakefiles/secrets.rb

```diff
@@ -77,45 +84,127 @@ def self.collect_secrets()
   def self.set_secrets(collected_secrets, exekube_cmd)
     return if collected_secrets.empty?
 
-    FileUtils.mkdir_p Secrets::SECRETS_DIR
-
     collected_secrets.each do |encryption_key, secrets|
-      sh_filter "#{exekube_cmd} secrets-fetch #{encryption_key}"
+      decrypted_secrets = fetch_secrets(encryption_key, exekube_cmd)
 
-      secrets_file = "./#{Secrets::SECRETS_DIR}/#{encryption_key}/#{Secrets::SECRETS_FILE}"
-
-      begin
-        if File.exist?(secrets_file)
-          decrypted_secrets = YAML.load(File.read(secrets_file))
-          decrypted_secrets.each do |secret_name, secret_value|
-            ENV["TF_VAR_#{secret_name}"] = secret_value
-          end
-        else
-          populated_secrets = {}
-          secrets.each do |secret_name|
-            if ENV[secret_name.upcase].to_s.empty?
-              secret_value = SecureRandom.hex
+      if decrypted_secrets
+        decrypted_secrets.each do |secret_name, secret_value|
+          ENV["TF_VAR_#{secret_name}"] = secret_value
+        end
+      else
+        populated_secrets = {}
+        secrets.each do |secret_name|
+          if ENV[secret_name.upcase].to_s.empty?
+            if secret_name.start_with?("key_")
+              key = OpenSSL::Cipher.new("aes-256-cfb").encrypt.random_key
+              secret_value = Base64.encode64(key).chomp
             else
-              secret_value = ENV[secret_name.upcase]
+              secret_value = SecureRandom.hex
             end
-            ENV["TF_VAR_#{secret_name}"] = secret_value
-            populated_secrets[secret_name] = secret_value
-          end
-          puts "Secret file '#{secrets_file}' not found. I will create one."
-          File.open(secrets_file, 'w') do |file|
-            file.write(populated_secrets.to_yaml)
+          else
+            secret_value = ENV[secret_name.upcase]
           end
-
-          sh_filter "#{exekube_cmd} secrets-push #{encryption_key}"
+          ENV["TF_VAR_#{secret_name}"] = secret_value
+          populated_secrets[secret_name] = secret_value
         end
-      rescue
-        puts "ERROR: Unable to populate secrets for key '#{encryption_key}'!"
-        raise
-      ensure
-        FileUtils.rm_f(secrets_file)
+
+        push_secrets(populated_secrets, encryption_key, exekube_cmd)
       end
     end
   end
+
+  def self.push_secrets(secrets, encryption_key, exekube_cmd)
+    gs_bucket = "#{ENV['TF_VAR_project_id']}-#{encryption_key}-secrets"
+    encoded_secrets = Base64.encode64(secrets.to_json).delete!("\n")
+
+    puts "[secret-mgmt] Encrypting secrets for key '#{encryption_key}'..."
+    encrypted_secrets = %x{
+      #{exekube_cmd} sh -c 'curl -s \
+      -H \"Authorization:Bearer $(gcloud auth print-access-token)\" \
```

Comment:
Is this curl call running inside docker image or is it calling docker?

Reply:
It is running from inside of the container, should probably remove `exekube_cmd` from there to reduce confusion.